codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,295 @@
+            )
+        )
+
+        # Get all the RUNNING jobs that are owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID (header won't be included with "--noheader" option)
+        # 1478828
+        # 1478830
+        running_jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '-t', 'RUNNING', '--format', '%A', '--noheader']
+        )
+        running_jobs = running_jobs.strip().split()
+
+        return [
+            WorkerJob(active=True) if job in running_jobs else WorkerJob(active=False)
+            for job in self.submitted_jobs
+        ]
+
+    def start_worker_job(self):
+        """
+        Start a CodaLab Slurm worker that submits batch job to Slurm
+        """
+        worker_id = uuid.uuid4().hex
+        # Set up the Slurm worker directory
+        slurm_work_dir = Path(self.args.slurm_work_dir, worker_id)
+        try:
+            slurm_work_dir.mkdir(parents=True, exist_ok=True)
+        except PermissionError as e:
+            logger.error(
+                "Failed to create the Slurm work directory: {}. "
+                "Stop creating a new worker.".format(e)
+            )
+            return
+
+        # This needs to be a unique directory since Batch jobs may share a host
+        worker_network_prefix = 'cl_worker_{}_network'.format(worker_id)
+        # Codalab worker's work directory
+        worker_work_dir = Path(
+            self.args.worker_work_dir_prefix, 'codalab-worker-scratch', worker_id
+        )
+        command = [
+            'cl-worker',
+            '--server',
+            self.args.server,
+            '--verbose',
+            '--exit-when-idle',
+            '--idle-seconds',
+            str(self.args.worker_idle_seconds),
+            '--work-dir',
+            str(worker_work_dir),
+            '--id',
+            worker_id,
+            '--network-prefix',
+            worker_network_prefix,
+            # always set in Slurm worker manager to ensure safe shutdown
+            '--pass-down-termination',
+        ]
+        if self.args.worker_tag:
+            command.extend(['--tag', self.args.worker_tag])
+        if self.args.password_file:
+            command.extend(['--password-file', self.args.password_file])
+
+        slurm_args = self.create_slurm_args(worker_id)
+        job_definition = self.create_job_definition(slurm_args=slurm_args, command=command)
+
+        # Not submit job to Slurm if dry run is specified
+        if self.args.dry_run:
+            return
+
+        batch_script = str(slurm_work_dir.joinpath(slurm_args['job-name'] + '.slurm'))
+        self.save_job_definition(batch_script, job_definition)
+        job_id_str = self.run_command([self.SBATCH, batch_script])
+
+        match = re.match(self.SBATCH_COMMAND_RETRUN_REGEX, job_id_str)
+        if match is not None:
+            job_id = match.group(1)
```

Comment:
Can you add a logging message here? Maybe something along the lines of `Submitted batch job with job id <jobid>`

Reply:
That's a good point. In fact, I am adding this logging information in the `run_command` function, so that we have more useful information logged.